userdata/xDynamicConfig.ps1

```diff
@@ -10,6 +10,10 @@ Configuration xDynamicConfig {
   Import-DscResource -ModuleName xWindowsUpdate
   Import-DscResource -ModuleName OpenCloudConfig
 
+  LocalConfigurationManager {
```

Comment:
Is LocalConfigurationManager a function being defined somewhere?

Reply:
LCM is the name of the process that runs DSC.
While I was implementing this patch, I noticed that there were runs of DSC occurring that were not wrapped in `StartTranscript`, `StopTranscript` commands within the system eventlogs or papertrail. This indicated that occasionally, the running DSC process was not one that was triggered by OCC. This is something I'd noticed many times over the life of OCC, but I never properly understood why this was happening. I mean, I knew that DSC runs itself (in order to ensure that the system is always in the "desired state"), but I didn't know how to control that behaviour. So, I did some research. I found some great documentation at: https://docs.microsoft.com/en-us/powershell/dsc/managing-nodes/metaconfig4
I learned that you can control the behaviour of when and how LCM runs DSC by setting the `ConfigurationMode` property. The default setting is `ApplyAndMonitor` which means LCM applies your DSC manifest and then monitors (on a schedule) to continuously maintain the desired state, reapplying manifest components whenever it determines (by validating the component) that it should be reapplied. For our use case, this is definitely not the desired behaviour., because we only want DSC to run between taskcluster tasks, at system startup.
It turns out that if we modify the `ConfigurationMode` property to have a value of `ApplyOnly`, we get exactly the behaviour we want and are expecting. LCM runs DSC only when triggered by OCC.